### ISSUE TITLE
Is there a way to control the aspect ratio of the navigation images?

### ISSUE THREAD

**Issue by HanHsuanWu** [hyperspy/hyperspy#3236]
#### Describe the functionality you would like to see.
EELSSpectrum.plot() doesn't seem to take argument like plt.imshow(extent). Is there a way to control the aspect ratio of navigation images?

**Comment by CSSFrancis** [comment]
@HanHsuanWu I think the navigation extent is dependent on the scales for the different axes. So you can set the axes scales to whatever to get a different ratio. 

The extent keyword is used to plot things based on the scales/offsets so it shouldn't be directly adjusted.

**Comment by HanHsuanWu** [comment]
@CSSFrancis It seems like changing the scale in axs_manager does not change the aspect ratio?

`
import hyperspy.api as hs
import matplotlib.pyplot as plt
import numpy as np

s = hs.signals.BaseSignal(np.random.random((10,20)))
s.T.axes_manager
s.T.plot()
s.axes_manager[0].scale=0.1
s.axes_manager
s.T.plot()
`

![image](https://github.com/hyperspy/hyperspy/assets/112028355/4bde0053-dc4b-4ba5-aeed-02931d1a320f)

**Comment by HanHsuanWu** [comment]
It seems like hs.plot.plot_images has an "aspect" argument.

**Comment by ericpre** [comment]
@HanHsuanWu do you mean the aspect ratio of the matplotlib figure the matplotlib axes, or both?
Setting aspect ratio in matplotlib is not as straightforward as we would expect and can be a bit confusing, maybe if you give an example of what you are trying to do with matplotlib only, then we can see what is possible to do with hyperspy?

**Comment by HanHsuanWu** [comment]
Sorry for the confusion guys. I mean matplotlib figure. The same as aspect in hs.plot.plot_images

hs.plot.plot_images(s,aspect=2)
![image](https://github.com/hyperspy/hyperspy/assets/112028355/98a1c892-f44d-442a-aff9-c4c8dbd1daca)
hs.plot.plot_images(s,aspect=0.5)
![image](https://github.com/hyperspy/hyperspy/assets/112028355/33b4475d-0949-4e0b-9d02-38e40d27d5d9)

**Comment by ericpre** [comment]
If you want to have non-square pixels, this is not possible when plotting signal, because the `aspect` and `extent` of matplotlib imshow are set in a way the pixel are square. The relevant part of the code is:https://github.com/hyperspy/hyperspy/blob/1d0971a4f1c468be61fb54b5f42f0d0248850c77/hyperspy/drawing/image.py#L245-L263

This is one of these things that would be worth revisiting/improving. What is your use case? Would you be interested in giving a go at it?

**Comment by HanHsuanWu** [comment]
@ericpre Yes of course. Let's see if I can come up with anything! Thanks for pointing out where the code is (I couldn't find it).
